examples/task.hpp

```diff
@@ -43,53 +63,65 @@ struct task {
     }
     std::variant<std::monostate, T, std::exception_ptr> data_{};
   };
-  struct promise_type : _promise_base, std::execution::with_awaitable_senders<promise_type> {
+  struct _promise : _promise_base, std::execution::with_awaitable_senders<_promise> {
     task get_return_object() noexcept {
-      return task(coro::coroutine_handle<promise_type>::from_promise(*this));
+      return task(coro::coroutine_handle<_promise>::from_promise(*this));
     }
     coro::suspend_always initial_suspend() noexcept {
       return {};
     }
-    final_awaitable final_suspend() noexcept {
+    _final_awaitable final_suspend() noexcept {
       return {};
     }
     void unhandled_exception() noexcept {
       this->data_.template emplace<2>(std::current_exception());
     }
+    friend void tag_invoke(std::execution::set_error_t, _promise&&, const std::exception_ptr&) noexcept {}
+    friend void tag_invoke(std::execution::set_done_t, _promise&&) noexcept {}
```

Comment:
This is super-clowny. In R2, the receiver queries are constrained to operate on receivers only. (Issue #271 is about that.) So to make it possible to issue receiver queries against the coroutine promise, it needs to satisfy the `receiver` concept.

I think I would prefer to relax the constraints on the receiver queries.